### PR TITLE
apache-arrow: Update to version 13.0.0

### PR DESCRIPTION
**Pull request by essandess**
* Update to version 13.0.0
* Add Python 311
* Delete Python 37

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 13.4.1 22F770820d arm64
Xcode 14.3.1 14E300c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: devel/apache-arrow/Portfile [hunk 1]
```diff
@@ -83,11 +83,6 @@ depends_lib-append \
                     port:zlib \
                     port:zstd
 
-# https://github.com/apache/arrow/pull/35046
-patchfiles-append   patch-io_util.diff
-# https://github.com/apache/arrow/issues/35833
-patchfiles-append   patch-absl-ver.diff
```
Comment (herbygillot) [inline devel/apache-arrow/Portfile:89]: Can you remove the actual patch files as well?

Comment (essandess) [reply devel/apache-arrow/Portfile:89]: Right, thanks. Done.

**Comment by ryandesign** [comment]
```
port lint py311-pyarrow:
Error: Unknown dependency: py311-pytest-lazy-fixture
```

**Comment by essandess** [comment]
> port lint py311-pyarrow:
> Error: Unknown dependency: py311-pytest-lazy-fixture

Oops, thanks. Now added.

**Comment by essandess** [comment]
This PR currently suffers from the issue described in https://trac.macports.org/ticket/67900.

**Comment by mascguy** [comment]
It looks like you have a potential syntax error in the portfile:

```
  Creating port index in /Users/runner/work/macports-ports/macports-ports/ports
  Failed to parse file devel/apache-arrow/Portfile: invalid command name "size"
```

**Pull request merged**